plenum/server/node.py

```diff
@@ -2411,13 +2422,109 @@ def doStaticValidation(self, request: Request):
 
         self.execute_hook(NodeHooks.POST_STATIC_VALIDATION, request=request)
 
+    def validateTaaAcceptance(self, request: Request):
+        ledger_id = self.ledger_id_for_request(request)
+
+        if not self.ledgerManager.ledger_info(ledger_id).taa_acceptance_required:
+            logger.trace(
+                "{} TAA acceptance passed for request {}: "
+                "not required for ledger id {}"
+                .format(self, request.reqId, ledger_id)
+            )
+            return
+
+        config_req_handler = self.ledger_to_req_handler.get(CONFIG_LEDGER_ID)
+        if not config_req_handler:
+            raise LogicError("Config request handler is missed")
+
+        taa = None
+        taa_data = config_req_handler.get_taa_data()
+        if taa_data is not None:
+            taa, taa_seq_no, taa_txn_time = taa_data
+
+        if not taa:
```

Comment:
In case of empty TAA, only the test is empty, but the version is not. So, `not taa` check is not enough here.

Reply:
updated